### PR TITLE
Officially split off from uber/hacheck

### PR DESCRIPTION
**Pull request by EvanKrall**
@bobtfish @jolynch y'all might like to look at this.

This merges our yelp branch into the master branch of our fork. I've filed a request with github support to officially split off from uber/hacheck.

### CODE REVIEW

**Comment by EvanKrall** [comment]
I also updated .travis.yml to use tornado versions that we actually care about.

**Comment by EvanKrall** [comment]
It's also important to note that some changes were committed to our master and not to the `yelp` branch. These are not shown in the diff, but if we start building packages from master, our new packages will get those changes.

See those changes here https://github.com/Yelp/hacheck/compare/42cc25514880c833f2be70f198a6b8761ffd9c8f...Yelp:master

File: hacheck/spool.py [hunk 1]
```diff
@@ -15,51 +59,72 @@ def configure(spool_root, needs_write=False):
...
+            return False, info_dict
     except IOError:
-        return True, {'service': service_name, 'reason': ''}
+        return happy_retval
 
 
 def status_all_down():
     """List all down services
 
     :returns: Iterable of pairs of (service name, dict of extra information)
```
Comment (jolynch) [inline hacheck/spool.py]: fix or TODO: let's update this docstring to take into account the interface changes

File: hacheck/spool.py [hunk 2]
```diff
@@ -15,51 +59,72 @@ def configure(spool_root, needs_write=False):
...
     else:
         return all_up, all_info
 
 
-def status(service_name):
+def status(service_name, port=None):
     """Check whether a service is asserted to be up or down, without checking
     the system-wide 'all' state.
 
     :returns: (bool of service status, dict of extra information)
```
Comment (jolynch) [inline hacheck/spool.py]: fix or TODO: let's update this docstring to take into account the interface changes (also what's in the extra information)

**Comment by jolynch** [comment]
👍 lg2m

Feel free to TODO the two comments I have, and let's fix itests in a follow up.

**Pull request merged**